compiler/passes/convert-uast.cpp

```diff
@@ -257,17 +376,102 @@ struct Converter {
   }
 
   BlockStmt* visit(const uast::Forall* node) {
-    INT_FATAL("TODO");
-    return nullptr;
+    if (node->isExpressionLevel()) {
+      INT_FATAL("TODO");
+      return nullptr;
+    } else {
+      INT_ASSERT(node->iterand());
+
+      // These are the arguments that 'ForallStmt::build' requires.
+      Expr* indices = nullptr;
+      Expr* iterator = toExpr(convertAST(node->iterand()));
+      CallExpr* intents = nullptr;
+      BlockStmt* body = createBlockWithStmts(node->stmts());
+      bool zippered = node->iterand()->isZip();
+      bool serialOK = false;
+
+      if (node->index()) {
+        indices = convertLoopIndexDecl(node->index());
+      }
+
+      if (node->withClause()) {
+        intents = toCallExpr(convertAST(node->withClause()));
+        INT_ASSERT(intents);
+      }
+
+      return ForallStmt::build(indices, iterator, intents, body, zippered,
+                               serialOK);
+    }
   }
 
   BlockStmt* visit(const uast::Foreach* node) {
     INT_FATAL("TODO");
     return nullptr;
   }
 
+  /// Non-Literal Initializer Expressions (e.g. Array, Range) ///
```

Comment:
This comment confused me a little, as I may have understood `Array` wrongly as a literal

Reply:
Suggestions? How about `/// Array, Range, Domain ///`?

Reply:
Yeah, something with less detail, I think. FWIW, this sounds like `var myArr = new array(1,2,3,4);` to me, which we don't have